clang/lib/Driver/SanitizerArgs.cpp

```diff
@@ -1099,7 +1099,13 @@ SanitizerArgs::SanitizerArgs(const ToolChain &TC,
     }
 
   } else {
-    AsanUseAfterScope = false;
+    if (AllAddedKinds & SanitizerKind::KernelAddress) {
```

Comment:
why just kernel?

Reply:
I see, there is Asan case already and we need kernel one

there is already one 
```
AsanUseAfterScope = Args.hasFlag(
        options::OPT_fsanitize_address_use_after_scope,
        options::OPT_fno_sanitize_address_use_after_scope, AsanUseAfterScope);
```

can you instead move that out from asas case and do single:


if (asan || kasan) {
AsanUseAfterScope = Args.hasFlag(
        options::OPT_fsanitize_address_use_after_scope,
        options::OPT_fno_sanitize_address_use_after_scope, AsanUseAfterScope);
} else {
AsanUseAfterScope = false
}